Changelogs for Digital Twin Manager Microservice 1025.1.0 (#4572)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/dtm-1025-1-0-improved-the-deletion-of-an-asset-to-not-remove-sub-assets-and-devices-that-are-still-used-in-other-hierarchies.md

---

> date: ""

Reviewer: ```suggestion
date: "2026-04-21"
```

> title: "Improved the deletion of an asset to not remove sub-assets and devices that are still used in other hierarchies"

Reviewer 2: ```suggestion
title: "Improved deletion of assets in hierarchies"
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-QHu1GdukP"
      label: "Feature"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2713"
version: "1025.1.0"
---
Previously, the deletion of an asset caused cascaded deletion of all

> sub-assets and assigned devices, even if they are still used in a

Reviewer 2: ```suggestion
subassets and assigned devices, even if they were still used in a
```

different asset or hierarchy. This could cause unwanted side effects to

> other parts of the asset hierarchy. This had been improved and now only

Reviewer 2: ```suggestion
other parts of the asset hierarchy. This behavior has been improved and now only
```

> sub-assets that are exclusively used in the hierarchy of the asset are

Reviewer 2: ```suggestion
subassets that are exclusively used in the hierarchy of the asset are
```

> removed. Other sub-assets will remain as is and the whole hierarchy will

Reviewer 2: ```suggestion
removed. Other subassets will remain as is, and the whole hierarchy will
```

> removed. Other subassets will remain as is, and the whole hierarchy will

Reviewer 2: ```suggestion
removed. Other subassets will remain unchanged, and the whole hierarchy will
```

stay intact.

> Furthermore a new query parameter `deleteDevices` is added to control

Reviewer 2: ```suggestion
Furthermore, a new query parameter `deleteDevices` has been added to control
```
